src/array.jl

```diff
@@ -212,6 +212,31 @@ end
 
             convert($A{T, N, R}, A)
         end
+
+        function Base.vcat{T,N,R}(A1::$A{T, N, R}, An::$A...)
+            As = (A1, An...)
+            levels = unique(T[[a.pool.levels for a in As]...;])
```

Comment:
Yes, should be good. Actually, the codes in `index` depend on the order of appearance in the input array, so even without calling `levels!` that test should already be OK (since values are not sorted in the original array for `a2`).